index.bs

```diff
@@ -245,6 +245,30 @@ partial interface NeuralNetworkContext {
 };
 </script>
 
+### concat ### {#api-neuralnetworkcontext-concat}
+<script type=idl>
+partial interface NeuralNetworkContext {
+  Operand concat(sequence<Operand> inputs, long dimension);
+};
+</script>
+
+<div algorithm=concat>
+    **Arguments:**
+        - *inputs*: a sequence of {{Operand}}. All input tensors must have the
+            same shape, except for the size of the dimension to concatenate on.
+        - *dimension*: a {{long}}. It specifies the dimension to concatenate on.
```

Comment:
I think the `dimension` argument should instead be named `axis`. It is a known term across many ML API. I think webnn should be consistent in this regard. Also since this won't be the first API that needs `axis`, what name we choose here will act as a precedence for the remaining of the API.

Reply:
To help with the description of the `axis` argument, here is a suggestion: "The axis that the inputs concatenate along, with the value in the interval [0, N), default to 0, where N is the rank of all the inputs."

Reply:
The `axis` name looks good. 

> default to 0

Do you mean it is an optional argument with default value as 0. It would change the signature to `Operand concat(sequence<Operand> inputs, optional long axis = 0);`. I am not aware any native ML APIs, e.g. ONNX or XLA-HLO, follow this. Could you please share some pointers? Thanks.

Reply:
I don't feel strongly about it being an optional argument. It's up to you. Just note that TF.js has axis as optional.

Reply:
Thanks for TF.js reference. As most APIs don't use default value 0, I'll follow that.

Reply:
Fixed in new commit. Please take a look.